src/source/PeerConnection/SessionDescription.c

```diff
@@ -85,23 +89,25 @@ STATUS deserializeSessionDescriptionInit(PCHAR sessionDescriptionJSON, UINT32 se
         } else if (STRNCMP(SDP_KEY, sessionDescriptionJSON + tokens[i].start, ARRAY_SIZE(SDP_KEY) - 1) == 0) {
             CHK((tokens[i + 1].end - tokens[i + 1].start) <= MAX_SESSION_DESCRIPTION_INIT_SDP_LEN,
                 STATUS_SESSION_DESCRIPTION_INIT_MAX_SDP_LEN_EXCEEDED);
-            last = curr = sessionDescriptionJSON + tokens[i + 1].start;
-            last -= ARRAY_SIZE(SESSION_DESCRIPTION_INIT_LINE_ENDING) - 1;
+            curr = sessionDescriptionJSON + tokens[i + 1].start;
             tail = sessionDescriptionJSON + tokens[i + 1].end;
             j = 0;
 
-            for (;curr <= tail; curr++) {
-                if (STRNCMP(curr, SESSION_DESCRIPTION_INIT_LINE_ENDING, ARRAY_SIZE(SESSION_DESCRIPTION_INIT_LINE_ENDING) - 1) == 0) {
-                    last += ARRAY_SIZE(SESSION_DESCRIPTION_INIT_LINE_ENDING) - 1;
-                    lineLen = (curr - last);
-
-                    MEMCPY((pSessionDescriptionInit->sdp) + j, last, lineLen);
-                    j += (lineLen + 1);
-                    pSessionDescriptionInit->sdp[j - 1] = '\n';
+            while ((next = STRNSTR(curr, SESSION_DESCRIPTION_INIT_LINE_ENDING_WITHOUT_CR, tail - curr)) != NULL) {
+                lineLen = (INT32) (next - curr);
 
-                    last = curr;
+                // there are escape characters before \n and \r, so we need to move back 1 more character
+                if (lineLen > 1 && curr[lineLen - 2] == '\\' && curr[lineLen - 1] == 'r') {
+                    lineLen -= 2;
                 }
-            }
+
+                MEMCPY((pSessionDescriptionInit->sdp) + j, curr, lineLen);
+                j += (lineLen + 2);
+                pSessionDescriptionInit->sdp[j - 2] = '\r';
+                pSessionDescriptionInit->sdp[j - 1] = '\n';
+
+                curr = next + 2;
```

Comment:
Ideally, you should add some comments describing that here you are skipping the source '\' 'n' characters. whereas in LN 105 you are adding 2 as you are adding \r and \n. Some readers might not have enough context on \r\n vs \n

Reply:
Ah, you're right. I'll add some comments.

Reply:
done